pipeline/extract_gog.py

```diff
@@ -0,0 +1,151 @@
+"""The extraction script for GOG"""
+import re
+from time import sleep
+import json
+import logging
+import requests
+from bs4 import BeautifulSoup
+from selenium import webdriver
+from selenium.webdriver.chrome.service import Service
+from webdriver_manager.chrome import ChromeDriverManager
+
+
+def get_soup(url: str) -> BeautifulSoup:
+    """Fetches the page content using Selenium and returns a BeautifulSoup object"""
+
+    options = webdriver.ChromeOptions()
+    # options.add_argument("--headless")
+    driver = webdriver.Chrome(service=Service(
+        ChromeDriverManager().install()), options=options)
+
+    driver.get(url)
+    sleep(3)
+    driver.execute_script("window.scrollTo(0, document.body.scrollHeight);") # Some resources only load when scrolling
+    sleep(3)
+    page_source = driver.page_source
+    driver.quit()
+
+    return BeautifulSoup(page_source, "html.parser")
+
+def scrape_newest(url: str) -> list[dict]:
+    """
+    Scrapes all the newest games from GOG games
+    """
+    response = requests.get(url)
+
+    soup = BeautifulSoup(response.text, features='html.parser')
+    game_links = [link['href'] for link in soup.find_all('a', href=True)
+                  if re.match(r'https://www\.gog\.com/en/game/', link["href"])]
+
+
+    page_data_list = []
+    for link in game_links:
+        game_data = get_data(link)
+        page_data_list.append(game_data)
+    return page_data_list
+
+
+def fetch_title(soup: BeautifulSoup):
+    """Gets the title of the page"""
+    title_tag = soup.find(class_='productcard-basics__title')
+    return title_tag.text.strip()
```

Comment:
This feels like a one line function call? 

```suggestion
title_tag = soup.find(class_='productcard-basics__title').text.strip()
```

Reply:
Error handling?